.github/workflows/auto_branching.yml

```
@@ -0,0 +1,214 @@
+### The auto-branching workflow triggered through a dispatch request from the CI
+name: auto-branching
+
+# Run on workflow dispatch from CI
+on:
+  workflow_dispatch:
+    inputs:
+      target_branch:
+        type: string
+        description: branch to be created from the master
+      stream_version:
+        type: string
+        description: new stream version of satellite
+
+jobs:
+  auto-branching-non-master:
+    name: ${{ github.event.inputs.target_branch }} - raise PR with changes
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create the ${{ github.event.inputs.target_branch }} branch
+        id: create-branch
+        uses: peterjgrainger/action-create-branch@v2.2.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: ${{ github.event.inputs.target_branch }}
+
+      - name: Create label for the ${{ github.event.inputs.target_branch }} branch
+        id: create-label
+        run: |
+          curl -X POST \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/labels \
+            -d "{\"name\":\"${{ github.event.inputs.target_branch }}\",\"color\":\"fbca04\"}"
+
+      - name: Switch to ${{ github.event.inputs.target_branch }} branch
+        run: git checkout -b "${{ github.event.inputs.target_branch }}"
+
+      - name: Checkout from ${{ github.event.inputs.target_branch }} branch for auto-branching changes
+        id: checkout-to-auto-branch
+        run: |
+            branch_name="auto-branching-${{ github.event.inputs.target_branch }}-$(date '+%s')"
+            git checkout -b "$branch_name"
+            echo "branch_name=$branch_name" >> $GITHUB_OUTPUT
+
+      - name: Update target branch label in dependabot yml file
+        id: update-dependabot
+        run: |
+            # Read the dependabot.yml file
+            FILE_PATH="./.github/dependabot.yml"
+            TARGET_BRANCH="${{ github.event.inputs.target_branch }}"
+
+            # Append the target branch label to the labels node
+            awk -v target="'$TARGET_BRANCH'" '/^ *labels:/ {$0 = $0 "\n      - " target} 1' "$FILE_PATH" > temp.yml && mv temp.yml "$FILE_PATH"
+
+      - name: Update repository URLs in requirements.txt
+        id: update-repo-urls
+        run: |
+            # Define the file path
+            FILE_PATH="./requirements.txt"
+
+            # Define the replacement strings
+            replacements=(
+              "git+https://github.com/SatelliteQE/airgun.git@master#egg=airgun git+https://github.com/SatelliteQE/airgun.git@${{ github.event.inputs.target_branch }}#egg=airgun"
+              "git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun git+https://github.com/SatelliteQE/nailgun.git@${{ github.event.inputs.target_branch }}#egg=nailgun"
```

The newer-style requirements will look like this. However, I think this will break the space-based delimeter below. Is there another delimiter you can use for the separation?

If needed, we can collapse the spaces here, but that is less readable in my opinion.
```suggestion
              "airgun @ git+https://github.com/SatelliteQE/airgun.git@master#egg=airgun airgun @ git+https://github.com/SatelliteQE/airgun.git@${{ github.event.inputs.target_branch }}#egg=airgun"
              "nailgun @ git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun nailgun @ git+https://github.com/SatelliteQE/nailgun.git@${{ github.event.inputs.target_branch }}#egg=nailgun"
```
ref: https://github.com/SatelliteQE/robottelo/pull/14399

---

sure will update

---

Done !

https://github.com/omkarkhatavkar/robottelo/actions/runs/8295064051/job/22701328471

https://github.com/omkarkhatavkar/robottelo/pull/266

https://github.com/omkarkhatavkar/robottelo/pull/267